inventory-sub/src/Item.java

```diff
@@ -0,0 +1,67 @@
+
+public class Item {
+    private String name;
+    private Integer price;
+    private Integer stock;
+    private String category;
+
+    public Item(String name, Integer price, Integer stock, String category) {
+        this.name = name;
+        this.price = price;
+        this.stock = stock;
+        this.category = category;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    public void setCategory(String category) {
+        this.category = category;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Integer getPrice() {
+        return price;
+    }
+
+    public void setPrice(Integer price) {
+        this.price = price;
+    }
+
+    public Integer getStock() {
+        return stock;
+    }
+
+    public void setStock(Integer stock) {
+        this.stock = stock;
+    }
+
+    public static Item createItem(String name, Integer price, Integer stock, String category) throws Exception {
+        if (category.equalsIgnoreCase("weapon")) {
+            return new Weapon(name, price, stock);
```

Comment:
More of a nitpicky thing but in general you want to have 1 `return` statement per method.  A better way would be to have a line that is `Item item` then in each if block do `item = Weapon(....)` or whatever the specific class is.